pkg/entity/batch_job/batch_job.go

```diff
@@ -59,29 +67,53 @@ func del(id *string, cmd *cobra.Command) {
 
 func create(cmd *cobra.Command) {
 	flags := cmd.Flags()
-	batchJobFile := util.GetStringAndErr(flags, batchJobsFileFlagName)
+	batchJobFile := util.GetStringAndErr(flags, batchJobFileFlagName)
+	batchJobType := getBatchJobType(flags)
 
-	batchJobData, err := ioutil.ReadFile(batchJobFile)
+	batchJobData, err := os.ReadFile(batchJobFile)
 	if err != nil {
 		common.CliExit(err)
 	}
 
+	var batchJobWrapper *entities.BatchJobWrapper
+	if batchJobType == encryptionType {
+		batchJobWrapper = createEncryptionBatchJob(cmd, batchJobData, flags)
+	} else if batchJobType == microAggregationType {
+		batchJobWrapper = createMicroAggregationBatchJob(cmd, batchJobData)
+	}
+	createBatchJobRequest := &batch_jobs.CreateBatchJobRequest{Job: batchJobWrapper}
+	response, err := client.CreateBatchJob(apiContext, createBatchJobRequest)
+	common.CliExit(err)
+
+	printer.Print(response)
+}
+
+func getBatchJobType(flags *pflag.FlagSet) string {
+	batchJobType := util.GetStringAndErr(flags, batchJobTypeFlagName)
+	if !(batchJobType == encryptionType || batchJobType == microAggregationType) {
+		common.CliExit(errors.New(fmt.Sprintf("Batch job type should be one of: %s, %s",
+			encryptionType, microAggregationType)))
+	}
+	return batchJobType
+}
+
+func createEncryptionBatchJob(cmd *cobra.Command,batchJobData []byte, flags *pflag.FlagSet) *entities.BatchJobWrapper {
 	batchJob := &entities.BatchJob{}
-	err = jsonpb.Unmarshal(bytes.NewReader(batchJobData), batchJob)
+	err := jsonpb.Unmarshal(bytes.NewReader(batchJobData), batchJob)
 	if err != nil {
 		common.CliExit(err)
 	}
 	projectId := project.GetProjectId(cmd)
 	batchJob.PolicyId = policy.GetPolicyFromFlags(flags)
-	setCommonProjectIds(batchJob, projectId)
-	createBatchJobRequest := &batch_jobs.CreateBatchJobRequest{BatchJob: batchJob}
-	response, err := client.CreateBatchJob(apiContext, createBatchJobRequest)
-	common.CliExit(err)
-
-	printer.Print(response)
+	setEncryptionBatchJobProjectIds(batchJob, projectId)
+	return &entities.BatchJobWrapper{
+		Job: &entities.BatchJobWrapper_EncryptionBatchJob{
+			EncryptionBatchJob: batchJob,
+		},
+	}
 }
 
-func setCommonProjectIds(batchJob *entities.BatchJob, projectId string) {
+func setEncryptionBatchJobProjectIds(batchJob *entities.BatchJob, projectId string) {
```

Comment:
Maybe inline this in the `createEncryptionBatchJob` fun.

Reply:
Why? I think it adds clutter and decreases readability of the create function.

Reply:
Ok! :)